wpilibj/src/main/java/edu/wpi/first/wpilibj/geometry/Pose2dSource.java

```diff
@@ -0,0 +1,62 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2020 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package edu.wpi.first.wpilibj.geometry;
+
+import java.util.function.Supplier;
+
+
+import edu.wpi.first.wpilibj.Sendable;
+import edu.wpi.first.wpilibj.kinematics.DifferentialDriveOdometry;
+import edu.wpi.first.wpilibj.smartdashboard.SendableBuilder;
+import edu.wpi.first.wpilibj.smartdashboard.SendableRegistry;
+
+/**
+ * A wrapper class of a <code>{@literal Supplier<Pose2d>}</code> and implements the {@code Sendable}
+ * interface, so the updated pose can be displayed on LiveWindow.
+ *
+ * <p>Can also be used as a <code>{@literal Supplier<Pose2d>}</code> in the RamseteCommand
+ * constructor.
+ */
+public class Pose2dSource implements Sendable, Supplier<Pose2d> {
+  private final Supplier<Pose2d> m_source;
+
+  /**
+   * Creates a {@code Pose2dSource} that wraps the given source.
+   * @param source the source of the poses, i.e the odometry's
+   * {@link DifferentialDriveOdometry#getPoseMeters() getPoseMeters()} method.
+   * @param name the name to associate with this Pose2dSource in LiveWindow.
+   Defaults to {@code "Robot Pose"}.
+   */
+  public Pose2dSource(Supplier<Pose2d> source, String name) {
+    m_source = source;
+
+    SendableRegistry.addLW(this, name);
+  }
+
+  /**
+   * Creates a {@code Pose2dSource} that wraps the given source.
+   * @param source the source of the poses, i.e the odometry's
+   * {@link DifferentialDriveOdometry#getPoseMeters() getPoseMeters()} method.
+   */
+  public Pose2dSource(Supplier<Pose2d> source) {
+    this(source, "Robot Pose");
```

Comment:
I don't think this should default to anything. Someone on the WPILib team might have a differing opinion on this however, so I would wait for someone to express theirs.

Reply:
Some teams, like ours, have never used LiveWindow, so forcing everyone to have a name for it doesn't make much sense if not everyone is going to use it.

Reply:
Technically, Shuffleboard also displays LW data, so I guess most teams implicitly use LW. But I totally agree with the fact that teams shouldn't be forced to chose a name.